src/common/TimeInterval.d.ts

```diff
@@ -16,4 +16,6 @@ export class TimeInterval {
 	public contains(other: TimeInterval): boolean;
 	public valueOf(): string;
 	public duration(specifier: string): number;
+	public getStartTimestamp(): moment.Moment;
```

Comment:
Not really a defect with this PR, but rather with TimeInterval; technically I think these are times, not timestamps, because their internal representation is not specified (it's just a Moment). Perhaps we should change the name?

Reply:
To me a time explicitly doesn't include a date (and it means that in python, postgres, ruby-on-rails, and other places). I like timestamp here, or datetime, because it conveys that we're tracking a moment in time as opposed to a moment in a day, and we have reasonably close to unix timestamp precision.

Reply:
Ah, that makes sense. Perhaps we should call it startMoment and endMoment, or just start and end? timeinterval.start and timeinterval.end reads pretty well, semantically.

Reply:
Start and end seem fine to me unless end is somehow reserved. I've been doing ruby for so long it seems a little weird.